pyro/util.py

```diff
@@ -342,3 +342,22 @@ def deep_getattr(obj, name):
     Throws an AttributeError if bad attribute
     """
     return functools.reduce(getattr, name.split("."), obj)
+
+
+def patch_dependency(target, root_module=torch):
```

Comment:
This introduces a cyclic dependency, since pyro.poutine depends on pyro.distributions depends on pyro.util depends on pyro.poutine. Could we keep this in pyro.distributions (which should not depend on anything outside of distributions), say in pyro.distributions.util or a new module pyro.distributions.patch?

Reply:
Thanks. I'm not sure how much of an issue it actually is, but I'd like to avoid any potential issues combining cyclic dependencies with hot patching :smile: